core/src/main/scala/org/apache/spark/executor/ExecutorMetricsPoller.scala

```diff
@@ -99,7 +99,7 @@ private[spark] class ExecutorMetricsPoller(
   def start(): Unit = {
     poller.foreach { exec =>
       val pollingTask: Runnable = () => Utils.logUncaughtExceptions(poll())
-      exec.scheduleAtFixedRate(pollingTask, 0L, pollingInterval, TimeUnit.MILLISECONDS)
+      exec.scheduleWithFixedDelay(pollingTask, 0L, pollingInterval, TimeUnit.MILLISECONDS)
```

Comment:
Similar like make GC from rate to delay.  It doesn't make sense to polling twice Immediately.